src/cider/nrepl/middleware/classpath.clj

```diff
@@ -1,16 +1,26 @@
 (ns cider.nrepl.middleware.classpath
   (:require [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
-            [orchard.misc :as u]
             [clojure.java.classpath :as cp]
-            [clojure.string :as str]))
+            [clojure.string :as str]
+            [orchard.misc :as u])
+  (:import [java.io File]))
+
+(def java-major-version
+  (-> (System/getProperty "java.version")
+      (str/split #"\.")
+      first
+      Long/parseLong))
 
 (defn classpath
   "Return a list of classpath entries for the current project.
   Takes into account the classpath trickery performed by Boot."
   []
   (if-let [classpath (u/boot-fake-classpath)]
-    (str/split classpath #":")
-    (map str (cp/classpath))))
+    (str/split classpath (re-pattern File/pathSeparator))
+    (if (< java-major-version 9)
```

Comment:
That's only a partial fix to the problem. Right now every call of `(cp/classpath)` is broken in our codebase (there are a few here and a few in `orchard`. I think the better fix would be to add some `classpath` ns in orchard and put there a wrapper around `cp/classpath` that's JDK 9 aware. There's also the question of `cp/classpath` not working well with boot, due to fake classpath there. Here we've patched this when we need strings, but I guess this doesn't work right when we need files.

Reply:
There are no calls to `clojure.java.classpath/classpath` outside of `cider.nrepl.middleware.classpath` in CIDER. So all the calls in CIDER do the right thing. More broadly, are you saying that `cider.repl.middleware.classpath` should be migrated to Orchard?

R.e. Boot, my understanding is that Boot builds its own classpath, so it's not affected by the Java 9 stuff. But I don't use Boot much so I could be wrong there. Ping @Deraen.

Reply:
> More broadly, are you saying that cider.repl.middleware.classpath should be migrated to Orchard?

Just this particular function. Things that are useful by themselves without middleware should go there, so we can later use them with CIDER and a socket REPL. That's why I started moving things to `orchard` in the first place. I wanted to decouple the middleware from some tooling API. With a socket REPL we'd simply be evaluating code directly over some dedicated connecting and processing the results for things that are middleware op invocations with nREPL. We're a long way from getting to the socket REPL support I guess, but that's one big step in that direction.